src/mgr/DaemonServer.cc

```diff
@@ -1353,39 +1357,65 @@ bool DaemonServer::_handle_command(
        }
     }
 
-    if (r && (prefix == "osd destroy" ||
-	      prefix == "osd purge")) {
+    if (prefix == "osd safe-to-destroy") {
+      if (!r) {
+        ss << "OSD(s) " << osds << " are safe to destroy without reducing data"
+           << " durability.";
+        safe_to_destroy.swap(osds);
+      }
+      if (f) {
+        f->open_object_section("osd_status");
+        f->open_array_section("safe_to_destroy");
+        for (auto i : safe_to_destroy)
+          f->dump_int("osd", i);
+        f->close_section();
+        f->open_array_section("active");
+        for (auto i : active_osds)
+          f->dump_int("osd", i);
+        f->close_section();
+        f->open_array_section("missing_stats");
+        for (auto i : missing_stats)
+          f->dump_int("osd", i);
+        f->close_section();
+        f->open_array_section("stored_pgs");
+        for (auto i : stored_pgs)
+          f->dump_int("osd", i);
+        f->close_section();
+        f->close_section(); // osd_status
+        f->flush(cmdctx->odata);
+        r = 0;
+        std::stringstream().swap(ss);
+      }
+      cmdctx->reply(r, ss);
+      return true;
+    }
+
+    if (r) {
       bool force = false;
       cmd_getval(cct, cmdctx->cmdmap, "force", force);
       if (!force) {
         // Backward compat
         cmd_getval(cct, cmdctx->cmdmap, "yes_i_really_mean_it", force);
       }
       if (!force) {
-	ss << "\nYou can proceed by passing --force, but be warned that this will likely mean real, permanent data loss.";
+        ss << "\nYou can proceed by passing --force, but be warned that"
+              " this will likely mean real, permanent data loss.";
       } else {
-	r = 0;
+        r = 0;
       }
     }
     if (r) {
       cmdctx->reply(r, ss);
       return true;
     }
-    if (prefix == "osd destroy" ||
-	prefix == "osd purge") {
-      const string cmd =
-	"{"
-	"\"prefix\": \"" + prefix + "-actual\", "
-	"\"id\": " + stringify(osds) + ", "
-	"\"yes_i_really_mean_it\": true"
-	"}";
-      auto on_finish = new ReplyOnFinish(cmdctx);
-      monc->start_mon_command({cmd}, {}, nullptr, &on_finish->outs, on_finish);
-    } else {
-      ss << "OSD(s) " << osds << " are safe to destroy without reducing data"
-	 << " durability.";
-      cmdctx->reply(0, ss);
-    }
+    const string cmd =
+      "{"
+      "\"prefix\": \"" + prefix + "-actual\", "
+      "\"id\": " + stringify(osds) + ", "
+      "\"sure\": \"--yes-i-really-mean-it\""
```

Comment:
this does not sound right. the key name should be `yes_i_really_mean_it`, not `sure`, and its value should be `true`, not `--yes-i-really-mean-it`.

see http://pulpito.ceph.com/kchai-2018-11-14_07:46:43-rados-wip-kefu-testing-2018-11-14-1049-distro-basic-smithi/

```
2018-11-14T16:52:51.006 INFO:tasks.workunit.client.0.smithi169.stderr:/home/ubuntu/cephtest/clone.client.0/qa/workunits/cephtool/test.sh:1316:
test_mon_osd_create_destroy:  ceph osd destroy osd.4 --yes-i-really-mean-it
2018-11-14T16:52:51.256 INFO:tasks.ceph.mgr.x.smithi169.stderr:2018-11-14 16:52:51.256 7f77a0ed1700 -1 mgr.server reply reply (1) Operation not
 permitted Are you SURE?  Did you verify with 'ceph osd safe-to-destroy'?  This will mean real, permanent data loss, as well as deletion of cep
hx and lockbox keys. Pass --yes-i-really-mean-it if you really do.
2018-11-14T16:52:51.257 INFO:tasks.workunit.client.0.smithi169.stderr:Error EPERM: Are you SURE?  Did you verify with 'ceph osd safe-to-destroy
'?  This will mean real, permanent data loss, as well as deletion of cephx and lockbox keys. Pass --yes-i-really-mean-it if you really do.
2018-11-14T16:52:51.271 INFO:tasks.workunit.client.0.smithi169.stderr:/home/ubuntu/cephtest/clone.client.0/qa/workunits/cephtool/test.sh:1: tes
t_mon_osd_create_destroy:  rm -fr /tmp/cephtool.boV
2018-11-14T16:52:51.273 DEBUG:teuthology.orchestra.run:got remote process result: 1
```

Reply:
@tchaikov Sorry :-(, fixed